app/src/lib/stores/app-store.ts

```diff
@@ -1780,6 +1780,40 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this._updateCurrentPullRequest(repository)
     this.updateMenuItemLabels(repository)
     this._initializeCompare(repository)
+    this.refreshRepositoryState([repository])
+  }
+
+  public refreshAllRepositories() {
+    return this.refreshRepositoryState(this.repositories)
+  }
+
+  private async refreshRepositoryState(
+    repositories: ReadonlyArray<Repository>
+  ): Promise<void> {
+    if (!enableRepoInfoIndicators()) {
+      return
+    }
+
+    const key = `refreshing ${repositories.length} repositories`
+
+    console.time(key)
+    for (const repo of repositories) {
+      await this.withAuthenticatingUser(repo, async (repo, account) => {
+        const gitStore = this.getGitStore(repo)
+        const lookup = this.localRepositoryStateLookup
+        await gitStore.fetch(account, true)
```

Comment:
@vanessayuenn and I talked a bunch about this part of the background work earlier - we're trying to balance two schools of thought here:

 - background fetching more frequently means that users can get feedback sooner
 - repositories with significant activity here means fetches will take longer (we have #1128 where users want to disable background fetching already)

I'm going to come back with some data to help quantify where we should end up on this spectrum

Reply:
So we currently have fetching enabled for every refresh, and using the Desktop repository as a reference this is the overhead:

 - macOS: `refreshing 1 repositories: 463.816162109375ms`
 - Windows: `refreshing 1 repositories: 1064ms`

This occurs whenever you switch back to Desktop from another application, or it can occur when switching repositories.

Every five minutes, the `setInterval` callback fires and all current repositories are refreshed (and fetched):

 - macOS: `refreshing 12 repositories: 5383.87890625ms`
 - Windows: `refreshing 9 repositories: 3438ms`

There are some complex repositories in the macOS list which is why it's now significantly worse, but the constraint still seems to be the `git fetch` operations themselves.

Reply:
Also note that I haven't factored in my internet connection speed (which is very good by world standards) - slower internet connections will likely take longer when fetching. This is offset by the fact this is being done in the background, but still a thing to keep in mind.

Reply:
>This occurs whenever you switch back to Desktop from another application, or it can occur when switching repositories.

@shiftkey I added a bit that skips `fetch` if the heavier `refreshRepository` operation has just been run less than 2 mins ago, which it should have everytime you switch repo and on application coming to focus.